### DISCUSSION TITLE
Version 2.3 Beta 1

### DISCUSSION THREAD

**Discussion by vkbo** [vkbo/novelWriter#1699] [General]
### Release Notes

This is a beta release of the next release version, and is intended for testing purposes. Please be careful when using this version on live writing projects, and make sure you take frequent backups.

Please check the changelog for an overview of changes. The full release notes will be added to the final release.

### Detailed Changelog

**Major Features**

* A new Welcome dialog has been added. The dialog replaces the Open Project dialog and the New Project Wizard. The Welcome dialog features artwork created by Louis Durrant, and a custom design for the project list. New projects can be created by a form available from the same dialog, and features a simplified set of options. Issue #1506. PRs #1647, #1681 and #1689.
* It is now possible to create a new project by copying the content of another project, or a Zip file of a project, including a backup. This option is available from the New Project feature of the Welcome dialog. Issue #841. PRs #1680 and #1684.
* The Preferences dialog has been completely redesigned. All options are now available in a single, scrollable list with appropriate section headers. All sections are available as navigation buttons along the side, and it is also possible to search for settings in a search box at the
  top. The design matches that created for the Manuscript Build Settings dialog added in 2.1. Issues #1603 and #1604. PR #1652.
* The Project Details dialog has been redesigned to match the other new dialogs. It has also been modified to properly handle multiple novel folders. The novel selector is placed at the top of the dialog, and affects all data in the tabs. The Novel Title info has been removed. PR #1665.
* The Manuscript Build Settings dialog has been updated to use the new config layout classes, which are more flexible in terms of content flow. PR #1674.
* A new root folder type called "Templates" has been added. Any document added here will show up in the Add Item menu in the Project Tree view under a "From Template" submenu. Selecting such an entry will create a new document at the selected location in the project, and populate it with the content of the template file. Issues #996 and #1125. PR #1688.
* The About novelWriter dialog has been simplified to only show some key information and the credits text. A link to the releases page is available for checking release notes. Keeping the release notes online means it is easier to update them, and make them more visually interesting as the formatting of the dialog box is limited. PR #1695.
* The old Check for Updates dialog has been removed. Checking for new releases of novelWriter can be done directly in the Welcome dialog or the About dialog by clicking "Check Now" next to the "Latest Version" label. PR #1696.

**Minor Features**

* A new drop down menu in the References panel below the Document Viewer has an option to filter out inactive notes in the various tag lists in the tabs on the panel. Issue #1653. PR #1654.
* The Novel Title fields, which no longer makes much sense after it was possible to add multiple novel folders to a project, has been fully dropped. Issue #1655. PR #1669.
* The document header for ODT manuscript files can now be customised. Issue #1505. PR #1675.
* The Manuscript Build tool can now insert characters (Point of View or Focus) into chapter headers and other headers. By default, it inserts the tag value, but the display name for a tag can be set with a `|` character in the `@tag` definition if a different text is desired. Issue #1468. PR #1677.
* A new modified comment format for the editor has been added. Instead of the regular comment using `%`, this one uses `%~`. The only difference is that the latter will never be exported to a manuscript at all, while regular comments can be exported when a setting is enabled. Issue #1075. PR #1690.
* It is now possible to change a document's label from the first heading in the document by right-clicking on it and selecting "Rename to heading". Issue #1443. PR #1692.
* The content of the Project Word List can now be exported and imported using plain text files. Issue #1560. PR #1691.
* The content of the Outline View can now be exported to a CSV file to be opened in any spreadsheet application. Issue #1507. PR #1697.

**Usability**

* A Create New submenu in the Project Tree context menu has been added, which give quick access to the items at the top level of the Add Item menu. Issue #1519. PR #1679.
* When multiple paragraphs are selected in the editor, and any of the comment features are toggled, or formatting is reset, the action is applied to all selected paragraphs, not just the first. Issue #1042 and #1687. PR #1690.

**Packaging**

* The in-app version format is now identical to the version tag, and the format for Ubuntu pre-release packages has been updated so that they are compatible with release packages. That is, if you add both the release and pre-release repos from Launchpad, release packages will now properly replace pre-release packages when running apt upgrade. PR #1659.

**Code Improvements**

* The Storage class has been refactored. It is the class that handles the project storage folder for a novelWriter project. The refactoring is a step towards allowing single file storage for projects as an alternative to project folders. Issue #977. PR #1635.
* All theme colours are now proper QColor objects from the start, which avoids the need to create a large number of these where they are used. PR #1656.
* A nwProject.bak file is no longer kept in the project folder. It never really served any purpose. The project file is still written to a temp file before the old file is replaced, which prevents partial overwrites. PR #1670.
* Other minor code improvements in PRs #1693 and 1694.

<hr /><em>This discussion was created from the release <a href='https://github.com/vkbo/novelWriter/releases/tag/v2.3b1'>Version 2.3 Beta 1</a>.</em>

**Comment by HeyMyian** [comment]
omg you're treating us! These changes are rad. I don't have capacities to check all of them out right now. But I did notice some UX issue on the new Welcome dialog when creating a New Project. The buttons for "Browse" and "Open" are misleading because the stuff these buttons refer to is not on the screen at that time.

I'm thinking the upper button row could replace the lower button row when on that screen. "Cancel" button should be retained for functionality. What do you think?

![Screenshot_2024-02-17_16-36-49](https://github.com/vkbo/novelWriter/assets/60368695/2c332242-36dc-474a-ae12-64f97b52d3cb)

**Reply by vkbo** [reply]
These are dialog buttons, and it is OS-dependant how they are laid out, so they need to be where they are, but I agree they should probably be disabled when in "New Project" mode. Could you make a ticket for it?

**Reply by HeyMyian** [reply]
Issue here: #1706

**Reply by vkbo** [reply]
I ended up making these changes last night. The downside is that the buttons will no longer conform to OS standards on the various platforms, which is what the previous implementation does. But I think that's fine on a very custom dialog anyway.

![Screenshot from 2024-02-21 12-39-12](https://github.com/vkbo/novelWriter/assets/1619840/e1c192f7-3de5-440d-bc81-db9af92b5ed9)

![Screenshot from 2024-02-21 12-39-16](https://github.com/vkbo/novelWriter/assets/1619840/4490d5e1-9921-4c2d-9272-42c535a04862)

**Comment by HeyMyian** [comment]
In the new Welcome dialog, when using "Copy an existing project" or "Create an example project", the slider knob things under "Project Notes" are correctly greyed out and inactive. The input boxes under "Chapter and Scenes" are also inactive, but there is no visual indication of their inactiveness, which there should be?

**Reply by vkbo** [reply]
That is controlled by the Qt framework, and they don't seem to have a different style when disabled. I have considered just making them invisible instead of disabled. That's probably better. Feel free to create an issue ticket so I can track it.

**Reply by HeyMyian** [reply]
Issue here: #1705

**Comment by HeyMyian** [comment]
In the Welcome dialog, it's not clear what the "Language" setting does. Upon further investigation, in the project settings it tells me this setting is only used when building the manuscript. Even there I don't actually know what it does.

From the Welcome dialog, I kind of assumed the Language setting would set the spellcheck language, or actually the UI language (since afaik the select options are the languages novelWriter has been translated into). It didn't do those things. I don't know what it does. :shrug:

**Reply by vkbo** [reply]
It's the build language. All text inserted into the manuscript by the app itself is localised, but the project may not be in the same language as the UI, so it's a separate setting. It's not a new setting. It's been there all along, but it used to be on the Build tool.

I'm not sure how to clarify it other than in the docs. It has nothing to do with spell checking either, because spell checking is an optional feature, but it wouldn't make sense to not set it to the same as spell checking.

**Reply by HeyMyian** [reply]
Ah, now I understand what it is, strings like `Comment:` in the compiled text. Why has it been taken out of the build tool? The build tool seems like a perfect place for it.

**Reply by vkbo** [reply]
It was moved to project settings mainly because there is no longer a single set of build settings like before. It does belong on the project level. In addition, the setting is applied to the document viewer as well, and I am planning to add direct exports from the project tree that will not go through the Manuscript tool, so it will be used in more places later. I should probably update the Project Settings help text.

The 2.x move to allow multiple Novel folders, and multiple builds, has caused a few things to shift around like that (like removing the novel title setting). Including planned features later in 2.x.

**Reply by vkbo** [reply]
I've thought a bit about this. I think I will remove this field from the New Project Form. Having it in Project Settings is enough. In most cases, it will be the same as the UI language, and I'll change the new project generator code to default to setting it to the UI language on fresh projects, and copy from template/example project when using the other options. The latter also makes more sense for template/copy. In project settings the distinction between manuscript language and spell checking is clearer.

**Reply by HeyMyian** [reply]
Sounds good!

**Comment by HeyMyian** [comment]
I'm loving the new illustration, it creates very cozy vibes. Well chosen 😊 ❤️

**Reply by vkbo** [reply]
Yeah, I love it too. I'm so happy with how it turned out. Louis did a great job. I had already seen his work and liked his style. I also had an idea what I wanted, and he brought that into reality just perfectly.
